test/compilable/test22997.d

```diff
@@ -0,0 +1,14 @@
+// https://issues.dlang.org/show_bug.cgi?id=22997
+
+struct Forward {}
+
+struct Foo
+{
+    this(scope ref typeof(this) rhs)
```

Comment:
Since `Foo` has no pointers, `scope` is stripped in type semantic. Is it relevant to the bug?

Reply:
Nope, I simply copy-pasted the provided bug report. I can strip it manually.